### PR TITLE
vector4 distance_squared_to and update csharp

### PR DESCRIPTION
**Pull request by antonWetzel**
### Add `Vector4::distance_squared_to` and update csharp to match https://github.com/godotengine/godot/pull/64027

- no tests for CSharp

### CODE REVIEW

**Review by aaronfranke — COMMENTED** [review]
The convention in the C# code is to keep instance methods alphabetized.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector4.cs [hunk 1]
```diff
@@ -178,16 +177,47 @@ public Vector4 Clamp(Vector4 min, Vector4 max)
             );
         }
 
+        /// <summary>
+        /// Performs a cubic interpolation between vector4 <paramref name="preA"/>, this vector,
```
Comment (aaronfranke) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Vector4.cs:181]:
```suggestion
        /// Performs a cubic interpolation between vectors <paramref name="preA"/>, this vector,
```

**Review by aaronfranke — APPROVED** [review]
Looks good aside from one documentation typo. I noticed that `DistanceSquaredTo` / `distance_squared_to` is missing from both core and C#, it should be easy to add if you want to do so here.

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**